### DISCUSSION TITLE
Sometimes content aint loading

### DISCUSSION THREAD

**Discussion by Joly0** [dani-garcia/vaultwarden#4859] [Q&A]
### Subject of the issue

I updated to the latest release 1.32 2 days ago. Now i wanted to login to my vaultwarden instance, just to notice, that nothing was there.
![grafik](https://github.com/user-attachments/assets/b21de367-5702-422b-8c10-53ef404529d1)
I dont see anything. After login the area thats supposed to show my entries loaded for quite a bit, and then stopped showing no entries.

In the devtools i see this error:
`[2024-08-14T19:27:45.893Z] Error: Connection disconnected with error 'Error: WebSocket closed with status code: 1006 (no reason given).'.`
And some time later i get this error message:
`NetworkError: A network error occurred.`
### Deployment environment

* vaultwarden version: 1.32

* Install method: Docker

* Clients used:  web vault, Android, Browser Extension

* Reverse proxy and version:  Nginx Proxy Manager

* MySQL/MariaDB or PostgreSQL version:  None?

* Other relevant details:

### Steps to reproduce
 I updated to 1.32 normally through unraid webui, it worked afterwards and it worked earlier this day, it now just stopped working.

### Expected behaviour

Should work and show my entries
### Actual behaviour

Doesnt show my entries

**Comment by Joly0** [comment]
Wow, right after writing this issue and after like 20 minutes of not working, it worked again. Though i am unsure what caused this issue, so maybe it should stay open. Also now noticed in my browser extension, that when i try to save an entry i get this:
![grafik](https://github.com/user-attachments/assets/e6f97c3a-0637-46cd-a277-3c15b9fad7c0)

**Comment by BlackDex** [comment]
Looks like either your internet, your server, or the reverse proxy had some connection issue for some reason.

**Reply by Joly0** [reply]
This is local access, only the reverse proxy was between because of local dns, but i even restarted the npm and it didnt solve the issue.
Not sure what on my server could have caused this otherwise

**Reply by BlackDex** [reply]
Then the reverse proxy could also be an issue. 
Validate the config of the reverse proxy, and also so see my remarks on the post below.

**Comment by blunweon** [comment]
I'm currently facing this problem but the difference is that I am able to add new entries through the web extension or desktop apps.

When entering the web-vault, there is a spinning circle (to denote loading?) that keeps on going for almost forever.
![floorp_1Bf7gXckUa](https://github.com/user-attachments/assets/02a67afc-ef81-4e44-8510-e516b3f0d3d4)

After awhile, it clears and shows the same thing that OP posted about. 

I had been using Vaultwarden with no issues from the past and suddenly it just stopped working.
Using SSH to get into the server hosting the Vaultwarden service,
`systemctl status vaultwarden` gives me a certain line `[rocket::server][WARN] Response indicated upgrade, but upgrade failed.` after which WS connection is (forced?) closed.

```
[response][INFO] (login) POST /identity/connect/token => 200 OK
[request][INFO] GET /api/sync?excludeDomains=true
[response][INFO] (sync) GET /api/sync?<data..> => 200 OK
[request][INFO] GET /api/accounts/revision-date
[response][INFO] (revision_date) GET /api/accounts/revision-date => 200 OK
[request][INFO] GET /notifications/hub?access_token=<redacted>
[vaultwarden::api::notifications][INFO] Accepting Rocket WS connection from 192.168.0.100
[response][INFO] (websockets_hub) GET /notifications/hub?<data..> => 200 OK
[rocket::server][WARN] Response indicated upgrade, but upgrade failed.
[vaultwarden::api::notifications][INFO] Closing WS connection from 192.168.0.100
```

**Reply by BlackDex** [reply]
A not working websocket connection isn't an issue and would not cause this. 

The only thing i can think of is a reverse proxy which is not configured correctly, also because the Websocket connection fails to work.

Several reasons could be there case. 
- caching (old web-vault files)
- security settings (ModSecurity/WAF/etc...)
- response/request manipulations like error response

Other firewall items like fail2ban or similar software.

**Reply by blunweon** [reply]
Thanks for the prompt reply.

Not sure what happened to my installation and borked it but I simply nuked my LXC and reinstalled Vaultwarden and voila - it worked without any edits or changes to my reverse proxy. This pushes me to think that it was due to caching or weird behaviour during updates that screwed up the entire system.

**Reply by BlackDex** [reply]
Strange indeed. But glad it works again.

**Comment by blacksheepaul** [comment]
Add `proxy_http_version 1.1;` if you are using nginx.

example: 
```nginx
 location / {
        proxy_pass http://vaultwarden;

        proxy_http_version 1.1;

        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection $connection_upgrade;

        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
    }
```

[refer](https://nginx.org/en/docs/http/websocket.html)
